spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/elasticsearch/sniff/SnifferBuilderCustomizer.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2012-2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.boot.autoconfigure.elasticsearch.sniff;
+
+import org.apache.http.client.config.RequestConfig;
+import org.apache.http.impl.nio.client.HttpAsyncClientBuilder;
+import org.elasticsearch.client.RestClientBuilder;
+import org.elasticsearch.client.sniff.SnifferBuilder;
+
+/**
+ * Callback interface that can be implemented by beans wishing to further customize the
+ * {@link org.elasticsearch.client.sniff.Sniffer} via a {@link org.elasticsearch.client.sniff.SnifferBuilder} whilst
+ * retaining default auto-configuration.
+ *
+ */
+@FunctionalInterface
+public interface SnifferBuilderCustomizer {
```

Comment:
We don't need this for a first pass. Those who want to configure the `NodesSniffer` can do so with their own `Sniffer` instance.